src/main/java/io/gravitee/risk/assessment/api/provider/RiskAssessmentProvider.java

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.gravitee.risk.assessment.api.provider;
+
+/**
+ * @author Rémi SULTAN (remi.sultan at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public interface RiskAssessmentProvider<PROPERTIES, RISK_ASSESSMENT> {
+    RISK_ASSESSMENT computeRiskAssessment(PROPERTIES properties);
```

Comment:
Should we return an optional ?

Reply:
Makes sense but I think this could make more sense to put it in the implementation ?

I always expect to return "something" (Assessment.NONE for instance)